test/unit/models/events.test.js

```diff
@@ -74,4 +76,117 @@ describe("Events", function () {
       }
     });
   });
+
+  describe("addPeerToEvent", function () {
+    it("should create a new peer document if it doesn't exist", async function () {
+      const docRef = await eventModel.add(eventData);
+
+      // Define sample peer data with a non-existing peer name
+      const peerData = {
+        name: "NonExistingPeer",
+        eventId: docRef.id,
+        role: "participant",
+        joinedAt: new Date(),
+      };
+
+      try {
+        const result = await eventQuery.addPeerToEvent(peerData);
+
+        // Get the created peer document from Firestore
+        const docSnapshot = await peerModel.doc(result.id).get();
+        const data = docSnapshot.data();
+
+        expect(data.name).to.equal(peerData.name);
+        expect(data.joinedEvents).to.have.lengthOf(1);
+        expect(data.joinedEvents[0].event_id).to.equal(peerData.eventId);
+        expect(data.joinedEvents[0].role).to.equal(peerData.role);
+      } catch (error) {
+        // Check that the function threw an error
+        expect(error).to.not.exist();
+      }
+    });
+
+    it("should update the joinedEvents array if the peer document exists", async function () {
+      const docRef = await eventModel.add(eventData);
+
+      // Define sample peer data with an existing peer name
+      const peerData = {
+        name: "ExistingPeer",
+        eventId: docRef.id,
+        role: "participant",
+        joinedAt: new Date(),
+      };
+
+      // Add an existing peer document to Firestore for the test
+      const existingPeerDoc = await peerModel.add({
+        name: peerData.name,
+        joinedEvents: [],
+      });
+
+      try {
+        await eventQuery.addPeerToEvent(peerData);
+
+        const docSnapshot = await peerModel.doc(existingPeerDoc.id).get();
+        const data = docSnapshot.data();
+
+        // Verify that the joinedEvents array was updated correctly
+        expect(data.joinedEvents).to.have.lengthOf(1);
+        expect(data.joinedEvents[0].event_id).to.equal(peerData.eventId);
+        expect(data.joinedEvents[0].role).to.equal(peerData.role);
+      } catch (error) {
+        // Check that the function threw an error
+        expect(error).to.not.exist();
+      }
```

Comment:
I think this catch block is unwanted as the test will fail if there is an error encountered even without this block

Reply:
Yes, fixed it.